pkg/cfn/manager/api.go

```diff
@@ -32,8 +32,8 @@ const (
 	resourceTypeAutoScalingGroup = "auto-scaling-group"
 	outputsRootPath              = "Outputs"
 	mappingsRootPath             = "Mappings"
-	ourStackRegexFmt             = "^(eksctl|EKS)-%s-((cluster|nodegroup-.+|addon-.+|fargate|karpenter)|(VPC|ServiceRole|ControlPlane|DefaultNodeGroup))$"
-	clusterStackRegex            = "eksctl-.*-cluster"
+	ourStackRegexFmt             = "^.*%s-((cluster|nodegroup-.+|addon-.+|fargate|karpenter)|(VPC|ServiceRole|ControlPlane|DefaultNodeGroup))$"
```

Comment:
I wonder what problems this could cause with existing stacks created by other users or instances.

The `.*-cluster` for example will match everything that is not eksctl but has -cluster in it. Including everything anyone created outside of eksctl.

Also, shouldn't this regex be dynamically created? For example, if disable is enabled, it's more relaxed, but otherwise, it's still `eksctl-.*-cluster`?

Reply:
Originally having two separate regexes based on the disable flag was the implementation but this goes back to @Callisto13's comment about UX and not wanting users to have to know whether the stack prefixing was disabled for every subsequent command past the create. If we don't want to force the user to let us know on a get/update/delete/etc if the stack prefixing was disabled, we need to check for both to cover each case every time.

As for matching on `-cluster`, that would still include the cluster name at that `%s` so it would limit to `.*-myclustername-cluster` which is much more narrow than `.*-cluster` thankfully. That said, I can tighten up the regex so it's only a triple of `(eksctl-|EKS-|)` as the prefixes to DescribeStack on, what do you think?

Reply:
I just realized that comment was about the line below (`clusterStackRegex`) instead of `ourStackRegexFmt`. That regex is only used for listing all clusters (`eksctl get cluster` without `--name`) so I feel like the UX for that should be that all clusters get returned regardless of prefixing instead of only having a list of un-prefixed clusters or only prefixed clusters. If there's a better way to reduce the false positives that `.*-cluster` would return without accidentally inducing false negatives I'm up for any ideas you might have!